src/ResourceManager/DataLakeStore/Commands.DataLakeStore/Commands/GetAzureRmDataLakeStoreItemContent.cs

```diff
@@ -16,86 +16,133 @@
         OutputType(typeof(byte[]), typeof(string))]
     [Alias("Get-AdlStoreItemContent")]
     public class GetAzureDataLakeStoreContent : DataLakeStoreFileSystemCmdletBase
     {
         private FileSystemCmdletProviderEncoding _encoding = FileSystemCmdletProviderEncoding.UTF8;
+        internal const string BaseParameterSetName = "Preview file content";
+        internal const string RowParameterSetName = "Preview file rows";
 
-        [Parameter(ValueFromPipelineByPropertyName = true, Position = 0, Mandatory = true,
+        [Parameter(ValueFromPipelineByPropertyName = true, Position = 0, ParameterSetName = BaseParameterSetName, Mandatory = true,
+            HelpMessage = "The DataLakeStore account to execute the filesystem operation in")]
+        [Parameter(ValueFromPipelineByPropertyName = true, Position = 0, ParameterSetName = RowParameterSetName, Mandatory = true,
             HelpMessage = "The DataLakeStore account to execute the filesystem operation in")]
         [ValidateNotNullOrEmpty]
         [Alias("AccountName")]
         public string Account { get; set; }
 
-        [Parameter(ValueFromPipelineByPropertyName = true, Position = 1, Mandatory = true,
+        [Parameter(ValueFromPipelineByPropertyName = true, Position = 1, ParameterSetName = BaseParameterSetName, Mandatory = true,
+            HelpMessage = "The path in the specified Data Lake account that should be read from. Can only be a file " +
+                          "in the format '/folder/file.txt', " +
+                          "where the first '/' after the DNS indicates the root of the file system.")]
+        [Parameter(ValueFromPipelineByPropertyName = true, Position = 1, ParameterSetName = RowParameterSetName, Mandatory = true,
             HelpMessage = "The path in the specified Data Lake account that should be read from. Can only be a file " +
                           "in the format '/folder/file.txt', " +
                           "where the first '/' after the DNS indicates the root of the file system.")]
         [ValidateNotNull]
         public DataLakeStorePathInstance Path { get; set; }
 
-        [Parameter(ValueFromPipelineByPropertyName = true, Position = 2, Mandatory = false,
+        [Parameter(ValueFromPipelineByPropertyName = true, Position = 2, ParameterSetName = RowParameterSetName, Mandatory = false,
+            HelpMessage =
+                "The number of rows (new line delimited) from the beginning of the file to preview. If no new line is encountered in the first 4mb of data, only that data will be returned."
+            )]
+        public long Head { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = true, Position = 3, ParameterSetName = RowParameterSetName, Mandatory = false,
+            HelpMessage = "The number of rows (new line delimited) from the end of the file to preview. If no new line is encountered in the first 4mb of data, only that data will be returned.")]
+        public long Tail { get; set; }
```

Comment:
This is a good idea. I will change it to do the validation of the range. In this case i will do so and also switch to ulong since you are right, these can always only be positive.